services/agents/registry.go

```diff
@@ -268,8 +268,36 @@ func (r *Registry) Run(stream agentpb.Agent_ConnectServer) error {
 					Payload: new(agentpb.ActionResultResponse),
 				})
 
+			case *agentpb.JobResult:
+				if e := r.db.InTransaction(func(t *reform.TX) error {
+					res, err := models.FindJobResultByID(t.Querier, p.JobId)
+					if err != nil {
+						return err
+					}
+
+					switch r := p.Result.(type) {
+					case *agentpb.JobResult_Error_:
+						res.Error = r.Error.Message
+					case *agentpb.JobResult_Echo_:
+						if res.Type != models.Echo {
```

Comment:
I think we should move result handling outside of registry to `JobsService` or `JobsRegistry`

Reply:
Done